common/errors.go

```diff
@@ -1567,6 +1567,20 @@ var NewErrEndpointMissingData = func(cause error) error {
 	}
 }
 
+type ErrEndpointDataOutOfRangeForFullNode struct{ BaseError }
+
+const ErrCodeEndpointDataOutOfRangeForFullNode = "ErrCodeEndpointDataOutOfRangeForFullNode"
+
+var NewErrEndpointDataOutOfRangeForFullNode = func(cause error) error {
+	return &ErrEndpointDataOutOfRangeForFullNode{
+		BaseError{
+			Code:    ErrCodeEndpointDataOutOfRangeForFullNode,
+			Message: "remote endpoint data is out of range for full node",
+			Cause:   cause,
+		},
+	}
+}
```

Comment:
* I used "Endpoint" in errors that are returned from node itself (not a logic within erpc itself). So "Upstream" is probably a better one as it's because erpc itself thinks this node type is not good one (without even sending a request).
* If we use NodeTypeMismatch error, then we can have one or two fields in "details" of the error like (expectedNodeType: "archive" actualNodeType: "full") then this same error can be used in other places
* The word "Code" doesn't need to appear in the code string
```suggestion
type ErrUpstreamNodeTypeMismatch struct{ BaseError }

const ErrCodeUpstreamNodeTypeMismatch = "ErrUpstreamNodeTypeMismatch"

var NewErrUpstreamNodeTypeMismatch = func(cause error, expected EvmNodeType, actual EvmNodeType) error {
	return &ErrUpstreamNodeTypeMismatch{
		BaseError{
			Code:    ErrCodeEndpointDataOutOfRangeForFullNode,
			Message: "remote endpoint data is out of range for full node",
			Cause:   cause,
			Details: // ... {expectedNodeType: expected, ...
		},
	}
}
```